time_calculation_inf.py

```diff
@@ -47,6 +42,59 @@ def _decode_node_breakdown(
             name="kv_cache_store",
         ) + self.O
 
+        seq_degree = self._sequence_parallel_degree()
+
+        comm_kind = "all_reduce" if seq_degree == 1 else "reduce_scatter"
+
+        ffn_dim = self.hidden_dim * self.ffn_mult if self.ffn_mult else self.ffn_dim
+        gemm_shapes = gemm_shapes or LLM_util.process_decode_gemm_shapes(
+            batch_size=batch_size,
+            current_seq_len=total_seq_len,
+            d_model=self.hidden_dim,
+            num_heads=self.num_heads,
+            kv_heads=self.kv_heads,
+            ffn_dim=ffn_dim,
+            vocab_size=self.vocab_size,
+            option="multiply_batch_into_m",
+        )
+
+        gemm_qkv_proj = gemm_shapes["qkv_proj"]
+        gemm_attention_score = gemm_shapes["attention_score"]
+        gemm_attention_output = gemm_shapes["attention_output"]
+        gemm_output_proj = gemm_shapes["output_proj"]
+        gemm_ffn1 = gemm_shapes["ffn1"]
+        gemm_ffn2 = gemm_shapes["ffn2"]
+
+        qkv_proj_time, _ = self._tensor_parallelism_gemm_forward(
+            gemm_qkv_proj, "decode_qkv_proj_f", tensor_type="column"
```

Comment:
didn't tensor_type = "row" and = "column" get deprecated? please fix inference code so that it works with the new additions.